omod/src/main/web/dashboardwidgets/latestobsforconceptlist/latestobsforconceptlist.controller.js

```diff
@@ -64,13 +46,29 @@ export default class LatestObsForConceptListController {
             '8d4a591e-c2cc-11de-8d13-0010c6dffd0f',
             '8d4a5af4-c2cc-11de-8d13-0010c6dffd0f'].indexOf(obs.concept.datatype.uuid) > -1) {
             //If value is date, time or datetime
-            var date = this.$filter('date')(new Date(obs.value), this.config.dateFormat);
-            return date;
+            return this.$filter('date')(new Date(obs.value), this.config.dateFormat);
         } else if (angular.isDefined(obs.value.display)) {
             //If value is a concept
-             return obs.value.display;
+             return this.getConceptName(obs.value, this.config.conceptNameType);
         } else {
             return obs.value;
         }
     }
+
+    getConceptName(concept, nameType) {
+        const names = concept.names.filter(n => !n.voided && n.locale === this.config.locale);
+        const fsn = names.filter(n => n.conceptNameType === "FULLY_SPECIFIED")[0];
+        const short = names.filter(n => n.conceptNameType === "SHORT")[0];
```

Comment:
are array index references (ie '[0]') null-safe in JavaScript?  (I'm guessing the answer is yes and that's why you did it this way).

Reply:
Not null-safe, but index-oob safe, yes (if the array is null it will throw; but if it is a length-zero array it will simply return `undefined`).